test/evergreen/code_complexity_analysis.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import ast
+from collections import defaultdict
+import csv
+from heapq import nlargest
+import json
+import os.path
+import pandas as pd
+
+def get_atlas_compatible_code_statistics(summaryFile, dataFile, outfile='atlas_code_complexity.json'):
+    ''' Generate the Atlas compatible format report. '''
+    atlasFormat = {
+                'Test Name': "Code Complexity",
+                'config': {},
+                'metrics': get_code_complexity(summaryFile, dataFile)
+            }
+
+    dirName = os.path.dirname(outfile)
+    if dirName:
+        os.makedirs(dirName, exist_ok=True)
+
+    with open(outfile, 'w') as outfile:
+        json.dump(atlasFormat, outfile, indent=4)
+
+def get_code_complexity(summaryFile, dataFile):
+    ''' Generate a list of intended contents from the complexity analysis. '''
+    resultList = []
+    complexityDict = {}
+    complexityDict['Average'] = get_average(summaryFile)
+    complexityDict['Stats Ranges'] = get_complexity_ranges_list(dataFile)
+    complexityDict['Top ' + str(topNRegions) + ' Regions'] = get_region_list(dataFile)
+    
+    resultList.append(complexityDict)
+    return (resultList)
+
+def get_region_list(dataFile):
+    ''' Retrieve topNRegions/functions with the highest cyclomatic complexity values. '''
+    top5 = dataFrame.nlargest(topNRegions, 'std.code.complexity:cyclomatic')
+    atlasFormat = {}
+    for index, row in top5.iterrows():
+        atlasFormat[row["region"]] = row["std.code.complexity:cyclomatic"]
+
+    return (atlasFormat)
```

Comment:
Just curious if the parentheses are necessary here.

Reply:
It is just a coding style I used in this script.